### PR TITLE
Update socket mode for SSH transport if it's not supported

### PR DESCRIPTION
**Pull request by Yingshun**
In modular daemon environment, remote mode is set to 'direct' by
default, which is not supported for SSH transport.

depends on: https://github.com/avocado-framework/avocado-vt/pull/2674
https://github.com/avocado-framework/avocado-vt/pull/2680
Signed-off-by: Yingshun Cui <yicui@redhat.com>

### CODE REVIEW

**Comment by Yingshun** [comment]
**Test result:**
```
 (1/1) type_specific.io-github-autotest-libvirt.virsh.migrate_vm.positive_testing.p2p_migration.bi_directional.without_postcopy: PASS (148.18 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 150.54 s
```

**Comment by Yingshun** [comment]
@dzhengfy Can you please help review it?

**Comment by chunfuwen** [comment]
@yafu-1 
Specify mode=direct in libvirt.conf is not mandatory in modular daemons testing, actually virsh client side can correctly connect modular socket rather than legacy libvirtd socket if  uri is specified. So we may not enable Specify mode=direct in libvirt.conf when do modular daemons testing.

Moreover, some references is available, https://libvirt.org/uri.html

**Comment by yafu-1** [comment]
> @yafu-1
> Specify mode=direct in libvirt.conf is not mandatory in modular daemons testing, actually virsh client side can correctly connect modular socket rather than legacy libvirtd socket if uri is specified(by default, uri is specified in virsh.py under avocado-vt). So we may not enable Specify mode=direct in libvirt.conf when do modular daemons testing.
> 
> Moreover, some references is available, https://libvirt.org/uri.html

Yes, it's not mandatory. If not specify mode=direct in libvirt.conf, 'virsh list' will connect 'qemu:///system' by virtproxyd, but virtproxyd may be not enabled in the custom env, in my understanding, the direct access to subdriver mode is the common use case.

**Comment by fangge1212** [comment]
Hi @dzhengfy @chunfuwen @Yingshun 
How is this pr going? I'm waiting for this pr merged to re-trigger the migration auto job(nearly 200 auto cases failed due to this issue).

**Comment by Yingshun** [comment]
> Hi @dzhengfy @chunfuwen @Yingshun
> How is this pr going? I'm waiting for this pr merged to re-trigger the migration auto job(nearly 200 auto cases failed due to this issue).

@fangge1212  Talked it with dzheng this morning, she will check it today. Thanks!

**Comment by dzhengfy** [comment]
Per @Yingshun , she did test for the changes from L2529 to L2544, but forgot to add the result in the comment. Now due to feature owner is waiting for this PR merged for several days, we decide to merge it first and paste more jenkins test result in this PR.

**Review by dzhengfy — APPROVED** [review]
LGTM

File: libvirt/tests/src/migration/migrate_vm.py [hunk 1]
```diff
@@ -2525,6 +2526,22 @@ def run(test, params, env):
...
+                remote_session = remote.wait_for_login('ssh', server_ip, '22',
+                                                       server_user, server_pwd,
+                                                       r"[\#\$]\s*$")
+                if utils_split_daemons.is_modular_daemon(remote_session):
+                    remote_dargs = {'server_ip': server_ip,
+                                    'server_user': server_user,
+                                    'server_pwd': server_pwd,
+                                    "file_path": "/etc/libvirt/libvirt.conf"}
+                    remote_libvirt_file = libvirt_config.remove_key_in_file(
+                        ["remote_mode"], remote_params=remote_dargs)
```
Comment (fangge1212) [inline libvirt/tests/src/migration/migrate_vm.py:2540]: There is no remove_key_in_file in module libvirt_config, but there is remove_key_in_conf, which name would you like to keep?

Comment (dzhengfy) [reply libvirt/tests/src/migration/migrate_vm.py:2540]: Fixed in https://github.com/autotest/tp-libvirt/pull/3024

**Pull request merged**